packages/application/src/application/use-cases/create-application/subscribe-application-submissions.service.ts

```diff
@@ -24,11 +24,12 @@ export async function subscribeApplicationSubmissions(container: Container) {
       for (const record of newRecords) {
         if (shouldProcessRecord(record, processedRecords)) {
           const command = mapRecordToCommand(record)
-          await commandBus.send(command)
           processedRecords.add(record.id)
+          await commandBus.send(command)
         }
       }
     } catch (error) {
+      processedRecords.clear()
```

Comment:
do we want everything to disappear if there's an intermittent fetch error? it seems like it would be more stable not to clear all records here?